domain/src/main/java/com/yapp/domain/formatter/AlarmDateTimeFormatter.kt

```
@@ -0,0 +1,184 @@
+
+    fun getFormattedEarliestUpcomingAlarmDeliveryTime(
+        alarms: List<Alarm>,
+        formats: DeliveryTimeFormats,
+        now: LocalDateTime = LocalDateTime.now(), // 기본값으로 현재 시간 사용
+    ): String {
+        val earliestAlarmDateTime = alarms
+            .filter { it.isAlarmActive }
+            .mapNotNull { alarm ->
+                try {
+                    calculateNextOccurrence(alarm.hour, alarm.minute, alarm.repeatDays, now)
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)